### ISSUE TITLE
252 timeout issue for www.thetrainline.com

### ISSUE THREAD

**Issue by qashahin** [macbre/phantomas#682]
Hi,

When typing in the following:

`phantomas --url https://www.thetrainline.com/`

I get `phantomas: (252) Timeout`. 

I have tried running against other websites and phantomas --url works very well however it appears to throw a 252 error for https://www.thetrainline.com/.

Can someone provide a solution as to how I can get around this please?

Many thanks,
QAShahin

**Comment by sridharangopal** [comment]
@qashahin try to remove the www and try again. I have had the problem before and its intermittent. Not sure why though.

**Comment by qashahin** [comment]
@sridharangopal Many thanks for your reply. I tried the URL by removing the www however that did not seem to work. I got the following output:

```
phantomas --url https://thetrainline.com
phantomas: (252) Timeout
phantomas v1.18.0 metrics for <>:

* requests: 0
* gzipRequests: 0
* postRequests: 0
* httpsRequests: 0
* notFound: 0
* bodySize: 0 bytes
* contentLength: 0 bytes
* httpTrafficCompleted: 0 ms
* timeToFirstByte: 0 ms
* timeToLastByte: 0 ms
* ajaxRequests: 0
* htmlCount: 0
* htmlSize: 0 bytes.....
...
...
* windowPrompts: 0
* bodyHTMLSize: 0 bytes

Offenders for jsErrors (7):
 * TypeError: undefined is not an object (evaluating 'window.__phantomas.log') - unknown fn: undefined @ 5 / unknown fn:  @ 8
 * TypeError: undefined is not an object (evaluating 'window.__phantomas.log') - unknown fn: undefined @ 6 / unknown fn:  @ 9
 * TypeError: undefined is not an object (evaluating 'new phantomas.nodeRunner') - unknown fn: undefined @ 3 / unknown fn:  @ 101 / unknown fn:  @ 102
 * TypeError: undefined is not an object (evaluating 'new phantomas.nodeRunner') - unknown fn: undefined @ 3 / unknown fn:  @ 65 / unknown fn:  @ 66
 * TypeError: undefined is not an object (evaluating 'phantomas.spyEnabled') - unknown fn: undefined @ 13 / unknown fn:  @ 56 / unknown fn:  @ 57
 * TypeError: undefined is not an object (evaluating 'phantomas.spyEnabled') - unknown fn: undefined @ 3 / unknown fn:  @ 59 / unknown fn:  @ 60
 * TypeError: undefined is not an object (evaluating 'phantomas.log') - unknown fn: undefined @ 14 / unknown fn:  @ 16 / unknown fn:  @ 17

Offenders for requestsWithTimeout (1):
 * https://thetrainline.com/
```

It looks like the URL did not capture any metrics when removing the `www`. I did try to run with and without `www` many times and the returned result was always 252 for both cases. I did try Google also (https://www.google.com and https://google.com) and both URL's worked therefore I don't think removing `www` causes an invalid URL as such. 

As mentioned I did try timeout and I to increase and decrease runs but sadly it always returned a 252. 

Is there any other possible solution.....

Many thanks in advance.

**Comment by sridharangopal** [comment]
@qashahin ... I just ran a test on your site, worked fine for me and I am able to see the analysis output. Must be a connectivity issue to the site on your end. Must be some firewall rules somewhere blocking you.
Try to change the UserAgent string to what a browser would send and then run the test to see if you can get through. This doe not look like a phantomas issue.

**Comment by qashahin** [comment]
@sridharangopal many thanks for taking the time to have a look, I really appreciate your help on this. Surprising I executed the same test this morning and made no changes to any of my scripts and it appears to work consistently. I did not change any firewall settings or anything else. 

I am glad that it now works but am also conscious that it my break again i.e. it now works and I have no idea why. Other than firewall rules (which have not changed) is there any other reason you believe that it was not working but is now?
